src/app/common/dialogs/export-pdf-dialog/export-pdf-dialog.component.ts

```diff
@@ -120,225 +191,280 @@ export class ExportPdfDialog extends BaseDialog {
 				);
 			}
 			this.emailsLoaded.then(async () => {
-				// image
-				const canvasWidth = 100;
-				const canvasHeight = 100;
-				const marginXImage = (pageWidth - canvasWidth) / 2;
-				let badge_img = new Image();
-				badge_img.src = badgeClass.image;
-				this.doc.addImage(badge_img, 'JPEG', marginXImage, yPos, canvasWidth, canvasHeight);
-
-				// title
-				yPos += canvasHeight;
-				this.doc.setFontSize(32);
-				this.doc.setFont('Helvetica', 'bold');
-				let title = this.doc.splitTextToSize(badgeClass.name, cutoff - this.doc.getTextWidth('...'));
-				let titlePadding = 0;
-				let maxTitleRows = 2;
-				if (title.length > maxTitleRows) {
-					title[maxTitleRows - 1] = title[maxTitleRows - 1] + '...';
-				} else if (title.length < maxTitleRows) {
-					titlePadding = (15 / 2) * (maxTitleRows - title.length);
-					yPos += titlePadding;
-				}
-				for (let i = 0; i < maxTitleRows; i = i + 1) {
-					if (title[i]) {
-						yPos += 15;
-						this.doc.text(title[i], pageWidth / 2, yPos, {
+				html2canvas(this.outputElement.nativeElement, {
+					backgroundColor: null,
+					scale: scaleFactor,
+				})
+					.then((canvas) => {
+						canvas.height = 100 * scaleFactor;
+						canvas.width = 100 * scaleFactor;
+						const image = this.generatePdfBackground(canvas);
+
+						const logoWidth = 50;
+						const logoHeight = 30;
+						const marginXImageLogo = 5;
+
+						let firstName = this.profile.firstName;
+						let lastName = this.profile.lastName;
+						firstName = firstName.charAt(0).toUpperCase() + firstName.slice(1);
```

Comment:
Could this cause problems if the first or last name has a length of 0?

Reply:
made an if check for the length of 0